Moonscraper Chart Editor/Assets/Scripts/Game/UI/Inspectors/NotePropertiesPanelController.cs

```diff
@@ -225,168 +247,72 @@ void Controls()
 	
     public void setTap()
     {
-        if (toggleBlockingActive)
-            return;
-
-        if (IsInNoteTool())
-        {
-            SetTapNoteTool();
-        }
-        else
-        {
-            SetTapNote();
-        }
+        SetNoteFlag(tapToggle, Note.Flags.Tap);
     }
 
-    void SetTapNoteTool()
+    public void setForced()
     {
-        if (tapToggle.interactable)
-            SetNoteToolFlag(ref noteToolController.desiredFlags, tapToggle, Note.Flags.Tap);
+        SetNoteFlag(forcedToggle, Note.Flags.Forced);
     }
 
-    void SetTapNote()
+    public void setCymbal()
     {
-        if (currentNote == prevNote)
-        {
-            var newFlags = currentNote.flags;
-
-            if (currentNote != null)
-            {
-                if (tapToggle.isOn)
-                    newFlags |= Note.Flags.Tap;
-                else
-                    newFlags &= ~Note.Flags.Tap;
-            }
-
-            SetNewFlags(currentNote, newFlags);
-        }
+        SetNoteFlag(cymbalToggle, Note.Flags.ProDrums_Cymbal);
     }
 
-    void SetNoteToolFlag(ref Note.Flags flags, Toggle uiToggle, Note.Flags flagsToToggle)
+    public void setDoubleKick()
     {
-        if ((flags & flagsToToggle) == 0)
-            flags |= flagsToToggle;
-        else
-            flags &= ~flagsToToggle;
+        SetNoteFlag(doubleKickToggle, Note.Flags.DoubleKick);
     }
 
-    public void setForced()
+    public void setAccent()
     {
-        if (toggleBlockingActive)
-            return;
-
-        if (IsInNoteTool())
-        {
-            SetForcedNoteTool();
-        }
-        else
-        {
-            SetForcedNote();
-        }
+        // A note can only be either an accent or a ghost, not both
+        SetNoteFlag(accentToggle, Note.Flags.ProDrums_Accent, Note.Flags.ProDrums_Ghost);
     }
 
-    public void setCymbal()
+    public void setGhost()
     {
-        if (toggleBlockingActive)
-            return;
-
-        if (IsInNoteTool())
-        {
-            SetCymbalNoteTool();
-        }
-        else
-        {
-            SetCymbalNote();
-        }
+        // A note can only be either an accent or a ghost, not both
+        SetNoteFlag(ghostToggle, Note.Flags.ProDrums_Ghost, Note.Flags.ProDrums_Accent);
     }
 
-    public void setDoubleKick()
+    void SetNoteFlag(Toggle toggle, Note.Flags flag, Note.Flags flagToExclude = Note.Flags.None)
     {
         if (toggleBlockingActive)
             return;
 
+        Note.Flags newFlags;
         if (IsInNoteTool())
         {
-            SetDoubleKickNoteTool();
+            if (toggle.interactable)
+                newFlags = noteToolController.desiredFlags;
+            else
+                return;
         }
```

Comment:
This doesn't look like it's updating the tool flags? Creates a copy of the tool flags rather than getting a reference.

Reply:
They get updated in SetNewFlags.
```
else
{
    // Updating note tool parameters and visuals
    noteToolController.desiredFlags = newFlags;
}
```